git_machete/github.py

```diff
@@ -44,69 +44,66 @@ def __parse_pr_json(pr_json: Any) -> GitHubPullRequest:
-    #
-    # TODO (#137): `gh` can store auth token for public and enterprise domains,
-    #  specify single domain for lookup.
-    # This is *only* github.com until enterprise support is added.
-    proc = subprocess.run(
-        [gh, "auth", "status", "--hostname", GITHUB_DOMAIN, "--show-token"],
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE
-    )
-
-    # gh auth status outputs to stderr in the form:
-    #
-    # {domain}:
-    #   ✓ Logged in to {domain} as {username} ({config_path})
-    #   ✓ Git operations for {domain} configured to use {protocol} protocol.
-    #   ✓ Token: *******************
-    #
-    # with non-zero exit code on failure
-    result = proc.stderr.decode()
-
-    match = re.search(r"Token: (\w+)", result)
-    if match:
-        return match.groups()[0]
-
-    return None
-
-
-def _get_token_from_hub() -> Optional[str]:
-    home_path: str = str(Path.home())
-    config_hub_path: str = os.path.join(home_path, ".config", "hub")
-    if os.path.isfile(config_hub_path):
-        with open(config_hub_path) as config_hub:
-            config_hub_content: str = config_hub.read()
-            # ~/.config/hub is a yaml file, with a structure similar to:
-            #
-            # {domain}:
-            # - user: {username}
-            #   oauth_token: *******************
-            #   protocol: {protocol}
-            match = re.search(r"oauth_token: (\w+)", config_hub_content)
-            if match:
-                return match.groups()[0]
+    def get_token_from_hub() -> Optional[str]:
+        home_path: str = str(Path.home())
+        config_hub_path: str = os.path.join(home_path, ".config", "hub")
+        if os.path.isfile(config_hub_path):
+            with open(config_hub_path) as config_hub:
+                config_hub_content: str = config_hub.read()
+                # ~/.config/hub is a yaml file, with a structure similar to:
+                #
+                # {domain}:
+                # - user: {username}
+                #   oauth_token: *******************
+                #   protocol: {protocol}
+                match = re.search(r"oauth_token: (\w+)", config_hub_content)
+                if match:
+                    return match.groups()[0]
 
-    return None
-
-
-def _get_token_from_env() -> Optional[str]:
-    return os.environ.get(GITHUB_TOKEN_ENV_VAR)
+        return None
 
+    def get_token_from_env() -> Optional[str]:
+        return os.environ.get(GITHUB_TOKEN_ENV_VAR)
 
-def _get_github_token() -> Optional[str]:
-    return _get_token_from_env() or __get_token_from_gh() or _get_token_from_hub()
+    return get_token_from_env() or get_token_from_gh() or get_token_from_hub()
 
 
-def __fire_github_api_request(method: str, url: str, token: Optional[str], request_body: Optional[Dict[str, Any]] = None) -> Any:
+def __fire_github_api_request(method: str, path: str, token: Optional[str], request_body: Optional[Dict[str, Any]] = None) -> Any:
```

Comment:
A little clarification, since it wasn't a full URL that was needed here, just the path